AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreObserveQueryTests.swift

```diff
@@ -104,11 +179,223 @@ class DataStoreObserveQueryTests: SyncEngineIntegrationTestBase {
+        deletePostAndWaitForSync(postMatchPredicateNoLongerMatches)
+        // Save "xyz 3" to force a snapshot to be emitted
+        savePostAndWaitForSync(Post(title: "xyz 3", content: testId, createdAt: .now()))
+
+        wait(for: [allSnapshotsReceived], timeout: 200)
+        sink.cancel()
+    }
+
+    /// ObserveQuery is set up with a sort order.
+    /// Sync is completed and actions are applied after sync to observe expected snapshots.
+    ///
+    /// - Given: DataStore has already synced the models to local store. ObserveQuery is started with a sort order
+    /// - When/Then:
+    ///    - Add several models with unique values for the field that is sorted on.
+    ///      The snapshot should have the models in the correct sorted order
+    ///    - Update models, modifying the field that is sorted on.
+    ///      The snasphot sould have the models in the correct sorted order
+    ///    - Delete models. The snapshot should have the models removed
+    ///
+    func testSortWithCreateUpdateDelete() throws {
+        let started = expectation(description: "Amplify started")
+        try startAmplify {
+            started.fulfill()
+        }
+        wait(for: [started], timeout: 2)
+        Amplify.Logging.logLevel = .info
+
+        let testId = UUID().uuidString
+        var snapshotCount = 0
+        let allSnapshotsReceived = expectation(description: "query snapshots received")
+        let sink = Amplify.DataStore.observeQuery(for: Post.self,
+                                                     where: Post.keys.content == testId,
+                                                     sort: .ascending(Post.keys.title))
+            .sink { completed in
+                switch completed {
+                case .finished:
+                    break
+                case .failure(let error):
+                    XCTFail("\(error)")
+                }
+            } receiveValue: { querySnapshot in
+                snapshotCount += 1
+
+                let items = querySnapshot.items
+                switch snapshotCount {
+                case 1:
+                    XCTAssertEqual(items.count, 0)
+                case 2, 3:
+                    XCTAssertEqual(items.count, 1)
+                    XCTAssertEqual(items[0].title, "a")
+                case 4, 5:
+                    XCTAssertEqual(items.count, 2)
+                    XCTAssertEqual(items[0].title, "a")
+                    XCTAssertEqual(items[1].title, "m")
+                case 6, 7:
+                    XCTAssertEqual(items.count, 3)
+                    XCTAssertEqual(items[0].title, "a")
+                    XCTAssertEqual(items[1].title, "m")
+                    XCTAssertEqual(items[2].title, "z")
+                case 8, 9:
+                    XCTAssertEqual(items.count, 3)
+                    XCTAssertEqual(items[0].title, "m")
+                    XCTAssertEqual(items[1].title, "n")
```

Comment:
can we assert on one more field like `content` to highlight that the actual `Post` object changed positions in the snapshot? 
for e.g. create posts with content like "1", "2", "3" and assert on content

Reply:
currently, the `content` is used to isolate data to the locally scoped models that were produced from this test, and then to only operate on the isolated data, this predicate is used `where: Post.keys.content == testId,` . If we modify the content then we won't have consistent count of items (items from other test runs will impact the latest test run). I can set the `rating` to 1.0, 2.0, 3.0 to achieve what you're asking but not sure why the assert on "m" and "n" is not sufficient?

Reply:
it was just to highlight that actual object(whose `title` was updated) changed places after sorting as other fields like `content` are identical for all posts. asserting on any other distinguishing field(like `id` or `rating`) would be a good to have. i would leave it up to you.